### PR TITLE
Fix Eventhubs vcpkg manifest

### PR DESCRIPTION
**Pull request by antkmsft**
Vcpkg CI currently fails: https://github.com/microsoft/vcpkg/pull/17119

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/vcpkg.json [hunk 1]
```diff
@@ -2,14 +2,7 @@
...
-      "name": "azure-core-cpp"
-    },
-    {
-      "name": "azure-core-amqp-cpp"
-    },
-    {
-      "name": "azure-storage-blobs-cpp"
-    }
+    "azure-core-amqp-cpp",
+    "azure-storage-blobs-cpp"
```
Comment (ahsonkhan) [inline sdk/eventhubs/azure-messaging-eventhubs/vcpkg.json:6]: This is an issue.
1) eventhubs should certainly not depend on storage blobs
2) Why aren't we listing core as a dependency? Is it coming transitively through the AMQP dependency?

File: sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json [hunk 2]
```diff
@@ -8,19 +8,18 @@
     "Microsoft Azure Messaging Event Hubs SDK for C++",
     "This library provides Azure Messaging Event Hubs SDK."
   ],
-  "homepage": "https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-keys",
+  "homepage": "https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/eventhubs/azure-messaging-eventhubs",
```
Comment (ahsonkhan) [inline sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json:11]: Good catch

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Overall looks good, but we need to find why we are depending on storage and remove that.

If that's outside the scope of this PR and requires more work to address, let me know, and I can file an issue instead.

**Comment by ahsonkhan** [comment]
> If that's outside the scope of this PR and requires more work to address, let me know, and I can file an issue instead.

Per offline discussion, this is working as intended for how EventHubs works, and hence a separate issue.

**Comment by ahsonkhan** [comment]
I was wondering what the failures/errors were in vcpkg.

One of them seems to be this:
https://dev.azure.com/vcpkg/public/_build/results?buildId=91832&view=logs&j=878666d5-db33-5b27-9e7d-b0c7ee352005&t=9a0acc60-e136-51ce-90f9-d151828126f7
> C:\a\1\s\ports\azure-messaging-eventhubs-cpp\vcpkg.json:17:36: error: Trailing comma in an object
    on expression:       "name": "azure-core-amqp-cpp",

I am not sure how we are fixing that here.

The other issue is:
https://dev.azure.com/vcpkg/public/_build/results?buildId=91832&view=logs&j=878666d5-db33-5b27-9e7d-b0c7ee352005&t=3980505e-92f5-53ef-75a8-4f2e70b97516
```text
2023-07-14T23:18:21.6352844Z ##[error]Found the following errors:
2023-07-14T23:18:21.6372669Z ##[error]versions/a-/azure-messaging-eventhubs-cpp.json(2,): error : azure-messaging-eventhubs-cpp is missing a version database file at version/a-/azure-messaging-eventhubs-cpp.json
```

File: sdk/eventhubs/azure-messaging-eventhubs/vcpkg.json (hunk 1 above)
Comment (antkmsft) [reply sdk/eventhubs/azure-messaging-eventhubs/vcpkg.json:6]: That is true - eventhubs depend on storage blobs.

**Comment by antkmsft** [comment]
@ahsonkhan, please re-review in the light of the fact that the dependency is correct.

File: sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json [hunk 3]
```diff
@@ -8,19 +8,18 @@
...
   "license": "MIT",
   "dependencies": [
-    {
-      "name": "azure-core-cpp",
-      "default-features": false,
-      "version>=": "1.9.0"
-    },
```
Comment (ahsonkhan) [inline sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json:18]: I would have expected to keep this and avoid transitive dependencies. Why not keep this?

Isn't this contributing to the issue:
https://github.com/Azure/azure-sdk-for-cpp/issues/4795

**Review by ahsonkhan — APPROVED** [review]

File: sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json (hunk 3 above)
Comment (antkmsft) [reply sdk/eventhubs/azure-messaging-eventhubs/vcpkg/vcpkg.json:18]: That's true, but I am bringing everything to the current way of doing things, and then when fixing another item, will fix the issue everywhere uniformly.

**Pull request merged**